actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -77,7 +95,7 @@ class ActionCable.Connection
       {identifier, message, type} = JSON.parse(event.data)
       switch type
         when message_types.welcome
-          @monitor.recordConnect()
+          break
```

Comment:
A couple reasons:
1. That `@monitor.recordConnect()` is in a place which makes it protocol dependent. As I recall you wanted to get rid of the welcome message anyways.
2. It was creating a race condition where it was called after the decision to disconnect had been made when there was a protocol mismatch (because the welcome message was sent right after the server opens the connection), but before there was an actual disconnect. Maybe this doesn't matter because the monitor is being shut off anyways, but it seemed like an undesirable behavior.